core/src/main/java/org/apache/accumulo/fate/zookeeper/ZooReaderWriter.java

```diff
@@ -49,13 +49,14 @@
   public ZooReaderWriter(AccumuloConfiguration conf) {
     this(conf.get(Property.INSTANCE_ZK_HOST),
         (int) conf.getTimeInMillis(Property.INSTANCE_ZK_TIMEOUT),
-        conf.get(Property.INSTANCE_SECRET));
+        conf.get(Property.INSTANCE_SECRET), ZooReader.DISABLED_RETRY_FACTORY);
```

Comment:
The behavioral differences between the constructors is unintuitive. Can we instead use a well-named static factory method, instead of the constructor to construct a ZRW for the specific intended purpose, so the constructor isn't unintentionally used by other code, and subject to the non-retrying behavior accidentally?

```java
// e.g.
ZooReaderWriter zrw1 = ZooReaderWriter.retrying(conf);
ZooReaderWriter zrw2 = ZooReaderWriter.nonRetrying(conf);
// or
ZooReaderWriter zrw = ZooReaderWriter.forXuseOnly(conf);
```